src/aiai_eval/task.py

```diff
@@ -641,6 +664,55 @@ def _adjust_label_ids(
         # TODO: Only a placeholder for now
         return model
 
+    def _prepare_carbon_tracker(
+        self,
+    ) -> Union[EmissionsTracker, OfflineEmissionsTracker]:
+        """Prepares a carbon emissions tracker.
+
+        Returns:
+            EmissionsTracker or OfflineEmissionsTracker:
+                A carbon emissions tracker. OfflineEmissionsTracker is returned if no internet
+                connection is available.
+        """
+        tracker_name = self.task_config.dataset_name
+        if self.evaluation_config.verbose:
+            log_level = "info"
+        else:
+            log_level = "error"
+
+        if internet_connection_available():
+            carbon_tracker = EmissionsTracker(
+                project_name=tracker_name,
+                measure_power_secs=5,
+                save_to_file=False,
+                save_to_api=False,
+                save_to_logger=False,
+                log_level=log_level,
+            )
+        else:
+            # If country_iso_code is "DNK", send warning
+            country_iso_code = self.evaluation_config.country_iso_code
+            if country_iso_code == "DNK":
+                message = (
+                    "The carbon tracker calculates carbon usage based on power consumption, "
+                    "and the country where the compute infrastructure is hosted. Internet connection "
+                    "was not available and hence the location of the infrastructure could not be "
+                    "automatically fetched, because of the this the location defaults to 'DNK'."
+                    "If the compute infrastructure is not hosted in Denmark, change the"
+                    "'country_iso_code' to the correct ISO code."
+                )
```

Comment:
If we change the default country code then this check should be changed accordingly.